### PR TITLE
Make selection() compatible with Libsass

### PR DESCRIPTION
**Pull request by kaishin**
This should address #615

### CODE REVIEW

**Comment by joshuaogle** [comment]
Why does Libsass have to be such a party pooper?

This is a clearer version of what we had before. Looks great :+1:

**Comment by felixbuenemann** [comment]
This also fixes compatibility with sass 3.3.

**Comment by gosukiwi** [comment]
This is great! Thanks a lot. Hopefully it gets merged soon :)

**Comment by tysongach** [comment]
:+1: 

This breaks current usage when using it on a selector (not a root level), though, so it’d have to wait until Bourbon 5.0.0. Bummer. How do we feel about making an exception and putting it with 4.2.0?

File: app/assets/stylesheets/css3/_selection.scss [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
+    &::selection {
+      @content;
+    }
 
-  #{$before-colons}::selection {
-    @content;
-  }
-  #{$before-colons}::-moz-selection {
-    @content;
+    &::-moz-selection {
```
Comment (tysongach) [inline app/assets/stylesheets/css3/_selection.scss]: The `-moz-` prefixes should go before the non-prefixed versions.

File: app/assets/stylesheets/css3/_selection.scss [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+      @content;
+    }
+  } @else {
```
Comment (tysongach) [inline app/assets/stylesheets/css3/_selection.scss]: Should we have one empty line above this?

**Comment by javierjulio** [comment]
Great to hear this fixes issue with sass 3.3! Thanks @kaishin. Looking forward to the next release.

**Pull request merged**